zip-proportional-output-fee-mechanism-pofm.rst

```diff
@@ -0,0 +1,242 @@
+This has resulted in high output transactions with 1,100 outputs costing the same as 
+transactions with 2 outputs.
+
+
+Requirements for gathering consensus
+------------------------------------
+
+Wallet developers SHOULD update the fees to the proposed formula under Specification section
+by Madars Virza and Kris Nuttycombe [#madars-1]_ 
+
+Where #inputs and #outputs also take into account transparent inputs and outputs. 
+Otherwise, the fee structure (if not otherwise changed) will preferentially encourage 
+usage of the transparent part of the protocol.
+
+The change to the conventional transaction fees must be undertaken soon
+as the Zcash network has been under heavy load with high-output transactions while 
+regular shielded transactions with 2 outputs are affected when relying on current 
+light wallet infrastructure.
+
+The following parties need to be part of the consensus:
+
+* The technical aspects of a changing conventional fee based on outputs 
+need to be evaluated.
+* A guarantee from mining groups is required to include the updated POFM 
+transactions in the next block.
+* Wallet developers need to update the software to use the new fee.
+* Zcash documentation and community outreach must be undertaken to
+make the change known.
+
+
+Requirements for adoption
+-------------------------
+
+The change to the conventional transaction fees should be undertaken soon
+as it gets difficult to gain consensus with the growth in the network
+of wallets, exchanges, miners, and third parties involved.
+
+The following parties need to be part of the consensus:
+
+* Support from mining groups is required to include the updated conventional
+  fee transactions in the next block.
+* Wallet developers need to provide a commitment to update the software to use
+  the new fee.
+* Zcash documentation and community outreach must be undertaken to make the
+  change known.
+* Changes to librustzcash crates to:
+  * expose minimum recommended fee for a given number of recipients
+  * implement default fee equation and corresponding unit tests
+  * integration of such mechanism to Transaction Builder APIs
+* adopt `librustzcash` changes to mobile SDK repos:
+  * ZcashLightClientKit
+  * zcash-light-client-ffi
+  * zcash-android-wallet-sdk
+
+
+Specification
+=============
+
+Wallets implementing this specification will use a conventional fee in the form of 
+base_fee and marginal_fee.
+
+The proposal would end up selecting one of:
+* base_fee = 1000, marginal_fee = 250 in @nuttycom's proposal.
+* base_fee = 1000, marginal_fee = 1000 in @madars' proposal.
+* base_fee = 10000, marginal_fee = 2500 in @daira's proposal.
+* base_fee = 1000, marginal_fee = 1000 for Shielded, Shielding and De-shielding
+transactions, and base_fee = 10000, marginal_fee = 10000 for Transparent transactions 
+per @nighthawk24's proposal.
+
+And calculated per the following forumla:
+min_fee = base_fee + marginal_fee * max(0, #inputs + #outputs - 3)
```

Comment:
Can you add rationale for the constant `3` here? If an Orchard action is considered 1 input + 1 output, then "ordinary" 2-action Orchard transactions are at a disadvantage with respect to Sapling transactions, which could have only 1 input. In order to put Sapling and Orchard on equal footing for a 2-output transaction, this constant should probably be an even number, preferably `4` as that represents the smallest transaction that can produce change.

Reply:
Please review this comment https://github.com/zcash/zips/pull/631#discussion_r948459240

Reply:
@nighthawk24 That was prior to the introduction of the separate `base_fee` from the `marginal_fee` value. Originally, Madars' proposal was `1000 + 1000 * max (0, inputs + outputs - 4)`.

A "standard" 2-in, 2-out transaction under the current fee proposal (with Madars' rates) would cost 2000 ZAT, not 1000 ZAT using the `3` constant.

Reply:
My comment about changing this constant to 3 was in the context of the original equation. Changing the equation to be clearer was an alternative fix, not in addition to.

Reply:
Updated formula to min_fee = base_fee + marginal_fee * max(0, inputs + outputs - grace_window_size)
where `grace_window_size` = 4